### ISSUE TITLE
Project build exception when referencing a project with PerformanceCounters msbuild package

### ISSUE THREAD

**Issue by alex-training** [Particular/NServiceBus.Metrics.PerformanceCounters#53]
Hi @Particular/metrics-maintainers, @weralabaj , @bording 

I have 2 project in my solution: the one for hosting NServiceBus endpoint with metrics, the other for unit tests.

The project with tests is referencing the main project and for some reason can not be build.
Every time on build I'm getting an exception: 

`Could not copy the file "Tests\bin\NServiceBus.Metrics.PerformanceCounters\Powershell\CreateNSBPerfCounters.ps1" because it was not found.`

![11](https://user-images.githubusercontent.com/13032219/37800390-4d44f26a-2e2b-11e8-8c86-4dce7db18871.png)

Notice that actual CreateNSBPerfCounters.ps1 is located in the wrong folder:
`Tests\bin\NServiceBus.Metrics.PerformanceCounters\bin\NServiceBus.Metrics.PerformanceCounters\Powershell\CreateNSBPerfCounters.ps1`

To be honest I was not expecting any ps-files inside the project with tests at all.

I'm using:
JetBrains Rider 2017/ Visual Studio 2017
NServiceBus.Metrics.PerformanceCounters.MsBuild" version="2.0.0"

**Comment by bording** [comment]
@alex-training The reason you're seeing that is because the NServiceBus.Metrics.PerformanceCounters.MsBuild package is a development dependency that only needs to be referenced by the main project. However, by default, PackageReferences are transitive across project references. If you add `PrivateAssets="All"` to the PackageReference, that should solve that problem. 

However, if that's the only package you're referencing, then the unit test project won't get the references to all of the other packages that it needs. I suggest changing your references to look like the following:
```
  <ItemGroup>
    <PackageReference Include="NServiceBus" Version="6.4.3" />
    <PackageReference Include="NServiceBus.Metrics.PerformanceCounters" Version="2.0.0" />
    <PackageReference Include="NServiceBus.Metrics.PerformanceCounters.MsBuild" Version="2.0.0" PrivateAssets="All" />
  </ItemGroup>
```

**Comment by alex-training** [comment]
@bording 
The unit tests project references only the main project and doesn't contain any metrics nuget packages.

I can't find PackageReference definition inside the csproj.

Did you mean a format of .net core projects?

**Comment by bording** [comment]
@alex-training What csproj format are you using? Are you using PackageReferences or packages.config?

Given the problem you're having I assumed I assumed it had to be PackageReferences, because I'm not aware of a way to have the problem you're describing otherwise.

Do you have a repro project you could share so I can see exactly how you have things set up?

**Comment by alex-training** [comment]
@bording 

I'm using packages.config.

I'm going to create a sample

**Comment by alex-training** [comment]
@bording 

So, I attached a very small sample.

I made some tests and I must say that the issue is reproducible only in [JetBrains Rider IDE](https://www.jetbrains.com/rider/)

Also it is strange that IDE generates 2 times "bin" folder:

`bin\NServiceBus.Metrics.PerformanceCounters\bin\NServiceBus.Metrics.PerformanceCounters`

I feel like the issue could be related to the standard msbuild variables used while building the project.

[MetricsWithIssue.zip](https://github.com/Particular/NServiceBus.Metrics.PerformanceCounters/files/1839876/MetricsWithIssue.zip)

**Comment by alex-training** [comment]
Found that the issue happens only with enabled checkbox "Use ReSharper Build" in Settings.

Questions are:
1) Why IDE builds the main project always (checked/unchecked Resharper build) and not the one with tests
2) Why build of the unit test project places the scripts from a different project in the output directory
3) How to fix the issue (and be able to use ReSharper Build as it is used by default) ...

![rider](https://user-images.githubusercontent.com/13032219/37808287-4ccb30a2-2e52-11e8-857d-bd267b8f59aa.png)

**Comment by alex-training** [comment]
FYI @SimonCropp

**Comment by SimonCropp** [comment]
@alex-training missing some context here. is there something specific u need my help with?

**Comment by alex-training** [comment]
@SimonCropp I saw you made some changes to NServiceBus.Metrics.PerformanceCounters.MsBuild.targets so maybe you can help.

Shortly the build is broken when using JetBrains Rider IDE (I attached sample project)

**Comment by alex-training** [comment]
@bording , @SimonCropp The issue could be related to AddScriptsToGetCopyToOutputDirectoryItems target

**Comment by bording** [comment]
@alex-training I took a look at your sample. When using it in Visual Studio and Rider with ReSharper Build disabled, everything is working as intended.

I also tried a version of your sample using the newer SDK-style project format with PackageReferences instead of packages.config. Once again, that works in all scenarios except for when ReSharper Build is enabled in Rider.

From what I can tell, ReSharper Build is not properly respecting the fact that the MsBuild package should not apply to the Tests project. This is true for both packages.config and PackageReferences.

Based on this, there really isn't anything we can do. You'll need to talk to JetBrains about improving ReSharper Build's behavior if you want to continue using it.
